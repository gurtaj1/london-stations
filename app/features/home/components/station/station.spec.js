import React from 'react';
import { shallow } from 'enzyme';

import Station, { tubeLineColors, capitaliseEachWord } from 'features/home/components/station/station';
import fixture from 'features/home/components/station/station.fixture';

describe('home.components', () => {
  describe('Station', () => {
    const props = fixture;

    it('should export something', () => {
      expect(Station).toBeDefined();
    });

    it('should render without crashing', () => {
      const wrapper = shallow(<Station {...props} />);

      expect(wrapper).toMatchSnapshot();
    });

    describe('tubeLineColors', () => {
      it('should export an object', () => {
        expect(typeof tubeLineColors).toBe('object');
      });
    
      it('should contain at least one tube line color', () => {
        const keys = Object.keys(tubeLineColors);
        expect(keys.length).toBeGreaterThanOrEqual(1);
      });
    
      it('should contain only string keys', () => {
        const keys = Object.keys(tubeLineColors);
    
        keys.forEach(key => {
          expect(typeof key).toBe('string');
        });
      });
    });

    describe('capitaliseEachWord()', () => {
      it('should capitalise the first letter of each word in a string', () => {
        const string = 'this is a test string';
        const output = capitaliseEachWord(string);

        expect(output).toEqual('This Is A Test String');
      });

      it('should should replace "-" with " and "', () => {
        const string = 'this-that';
        const output = capitaliseEachWord(string);

        expect(output).toEqual('This And That');
      });

      it('should capitalise the first letter of a one word string', () => {
        const string = 'test';
        const output = capitaliseEachWord(string);

        expect(output).toEqual('Test');
      });
    });
  });
});
