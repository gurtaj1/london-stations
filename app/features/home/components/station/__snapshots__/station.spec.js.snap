// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home.components Station should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Station
    name="station-name"
    tubeLines={
      Array [
        "circle",
        "district",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="station-name"
        >
          <Title
            title="station-name"
          />
        </div>,
        <div
          className="lines-list"
        >
          <ListItem
            inlineStyleColor="#FFCC00"
            listItem="Circle"
          />
          <ListItem
            inlineStyleColor="#006633"
            listItem="District"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Title
            title="station-name"
          />,
          "className": "station-name",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "station-name",
          "nodeType": "function",
          "props": Object {
            "title": "station-name",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ListItem
              inlineStyleColor="#FFCC00"
              listItem="Circle"
            />,
            <ListItem
              inlineStyleColor="#006633"
              listItem="District"
            />,
          ],
          "className": "lines-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "circle",
            "nodeType": "function",
            "props": Object {
              "inlineStyleColor": "#FFCC00",
              "listItem": "Circle",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "district",
            "nodeType": "function",
            "props": Object {
              "inlineStyleColor": "#006633",
              "listItem": "District",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="station-name"
          >
            <Title
              title="station-name"
            />
          </div>,
          <div
            className="lines-list"
          >
            <ListItem
              inlineStyleColor="#FFCC00"
              listItem="Circle"
            />
            <ListItem
              inlineStyleColor="#006633"
              listItem="District"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Title
              title="station-name"
            />,
            "className": "station-name",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "station-name",
            "nodeType": "function",
            "props": Object {
              "title": "station-name",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ListItem
                inlineStyleColor="#FFCC00"
                listItem="Circle"
              />,
              <ListItem
                inlineStyleColor="#006633"
                listItem="District"
              />,
            ],
            "className": "lines-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "circle",
              "nodeType": "function",
              "props": Object {
                "inlineStyleColor": "#FFCC00",
                "listItem": "Circle",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "district",
              "nodeType": "function",
              "props": Object {
                "inlineStyleColor": "#006633",
                "listItem": "District",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
